### PR TITLE
Eslint and stylelint all .vue files

### PR DESCRIPTION
**Pull request by jonahtanjz**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [x] Code maintenance
- [ ] Others, please explain:

**Overview of changes:**
A follow-up PR to #1462. Enable eslint and stylelint for all `.vue` files in `vue-components`.

**Anything you'd like to highlight / discuss:**
1. There are a few `console.log`s used which eslint will throw a warning. Should I disable the checks for these `console.log`s or remove them?
2. Use of `v-html` in some `.vue` files is also throwing warnings during the check. Should this be change from `<span v-html="html"></span>`  to `<span>{{ html }}</span>` instead to follow eslint rules? 

**Testing instructions:**
`npm run test`

**Proposed commit message: (wrap lines at 72 characters)**
Eslint and stylelint all .vue files

Currently, most .vue files are ignored during the linting checks.

Since these files are frequently modified, we should validate 
them in our ci tests to ensure they follow eslint rules. Furthermore, 
.vue files can contain raw css hence appropriate to also stylelint them.

Let's eslint and stylelint all .vue files so that they can be validated 
as well. 

---

**Checklist:** :ballot_box_with_check:

- [x] Updated the documentation for feature additions and enhancements
- [x] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No blatantly unrelated changes
- [x] Pinged someone for a review!

### CODE REVIEW

File: packages/vue-components/src/Tab.vue [hunk 1]
```diff
@@ -13,81 +16,82 @@
...
-      this.$parent.tabs.push(this)
+      this.$parent.tabs.push(this);
     }
   },
-  beforeDestroy () {
-    if (this._tabset.active === this.index) { this._tabset.active = 0 }
+  beforeDestroy() {
+    if (this._tabset.active === this.index) { this._tabset.active = 0; }
     if (this._ingroup) {
-      var index = parent.tabs.indexOf(this);
```
Comment (ang-zeyu) [inline packages/vue-components/src/Tab.vue:75]: 👀  should this have been referencing `$parent`?

Comment (jonahtanjz) [reply packages/vue-components/src/Tab.vue:75]: Yup seems like it should have been `$parent` as the function `parent.tabs.indexOf` does not exist.

**Review by ang-zeyu — APPROVED** [review]
Lgtm!

> There are a few console.logs used which eslint will throw a warning. Should I disable the checks for these console.logs or remove them?
> Use of v-html in some .vue files is also throwing warnings during the check. Should this be change from <span v-html="html"></span> to <span>{{ html }}</span> instead to follow eslint rules?

Thanks for the catch, I think we can resolve the second issue separately since it requires a substantial functional change.

For the console.logs, might be better to remove them as we shouldn't be relying on it to inform users of anything. Where appropriate, could replace them with comments for anyone taking a look at these files in the future.

**Comment by jonahtanjz** [comment]
@ang-zeyu Updated the PR to remove the consoles or replaced them with comments.

**Review by ang-zeyu — APPROVED** [review]

**Pull request merged**